treeherder/webapp/api/performance_data.py

```diff
@@ -423,3 +426,63 @@ class PerformanceIssueTrackerViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = IssueTrackerSerializer
     filter_backends = [filters.OrderingFilter]
     ordering = 'id'
+
+
+class PerformanceByRevision(generics.ListAPIView):
```

Comment:
Shouldn't this be a view set?

Reply:
Viewsets seemed to be designed for full CRUD API's with reusable logic but since this is only ever returning a read-only list, I think the ListAPIView makes sense here. Is there an advantage to using a Viewset (besides not having to configure urls)?

Reply:
Yes, I think you're right, re-reading the docs. I got thrown off by the fact that I (probably improperly) used ViewSets when generating some of the other Perfherder views.

Since this is a custom view rather than more-or-less a model serializer, you could also consider just using the generic APIView https://www.django-rest-framework.org/api-guide/views/

Reply:
It doesn't look like can define a serializer class with the APIView.